lib/jwt.js

```diff
@@ -66,6 +66,12 @@ jwt.decode = function jwt_decode(token, key, noVerify, algorithm) {
     throw new Error('Not enough or too many segments');
   }
 
+  // check noVerify - only accept a boolean value for noVerify to avoid user error
+  var typeofNoVerify = typeof noVerify;
+  if (typeofNoVerify !== 'undefined' && typeofNoVerify !== 'boolean') {
+    throw new Error('Invalid optional "noVerify" argument. It must be a boolean');
```

Comment:
IMHO, I think your error message should be shorter (like `'Invalid "noVerify" argument'`, or `'"noVerify" must be a boolean'`, then the user will have to check the documentation if they want to know more), and not a couple of sentences.

Errors are meant to be parsed at best, so I like to limit overhead (plus, punctuation might be confusing).

In this case, I'm not sure that error throwing should be required, but using `!==` is definitely better =)

That is merely suggestions, what do you think @hokaccha ?